haystack/nodes/prompt/prompt_node.py

```diff
@@ -187,65 +175,18 @@ def prompt(self, prompt_template: Optional[Union[str, PromptTemplate]], *args, *
                 results.extend(output)
         return results
 
-    def add_prompt_template(self, prompt_template: PromptTemplate) -> None:
-        """
-        Adds a prompt template to the list of supported prompt templates.
-        :param prompt_template: The PromptTemplate object to be added.
-        :return: None
-        """
-        if prompt_template.name in self.prompt_templates:
-            raise ValueError(
-                f"Prompt template {prompt_template.name} already exists. "
-                f"Select a different name for this prompt template."
-            )
+    @property
+    def default_prompt_template(self):
```

Comment:
Why do we need a property? Wouldn't it make more sense if we just keep it private?

Reply:
I tried not to break the API where it wasn't necessary. It used to be possible to set the default prompt template after initializing the promptnode, so this property should preserve that API.

If that's not important I'm happy to remove this functionality altogether, but lmk if in this PR or in another

Reply:
As much as I don't like the API it makes sense. Let's not break it for now.

Reply:
I'd make a change though. What do you think of renaming `_default_prompt_template` to just `_template`?

Reply:
Hm I'm on the fence on that. Technically PromptNode can have a default template and the template it's actually using for prompting, which is forgotten as soon as the call is over. Calling it just `_template` might be a bit confusing (is it the default? The one received from` .prompt()`? Does the one given to `.prompt()` get stored there? etc.)

Reply:
Right, the confusing API makes simple variables confusing. 😕 

I think we can remove the `prompt` part though, so `_default_template`?

Reply:
Sounds good! I agree the confusing API doesn't help writing simpler code :sweat_smile: